cloud: Add Airbyte doc in integrations section (#10823)
pingcap/docs: tidb-cloud/integrate-tidbcloud-with-airbyte.md

---
title: Integrate TiDB Cloud with Airbyte
summary: Learn how to use Airbyte TiDB connector.
---

# Integrate TiDB Cloud with Airbyte

[Airbyte](https://airbyte.com/) is an open-source data integration engine to build Extract, Load, Transform (ELT) pipelines and consolidate your data in your data warehouses, data lakes, and databases. This document describes how to connect Airbyte to TiDB as a source or a destination.

> ## Deploy Airbyte

Reviewer: ```suggestion
## Step 1. Deploy Airbyte
```

Airbyte supports local deployment with only a few steps.

1. Install [Docker](https://www.docker.com/products/docker-desktop) on your workspace.

2. Clone the Airbyte source code.

    ```shell
    $ git clone https://github.com/airbytehq/airbyte.git
    $ cd airbyte
    ```

3. Run the Docker images by docker-compose.

    ```shell
    $ docker-compose up
    ```

> Once you see an Airbyte banner, you can go to [http://localhost:8000](http://localhost:8000) to visit the UI.

Reviewer: ```suggestion
Once you see an Airbyte banner, you can go to <http://localhost:8000> to visit the UI.
```

> ![img](/media/tidb-cloud/integration-airbyte-ready.jpg)

Reviewer: ```suggestion
![Airbyte banner](/media/tidb-cloud/integration-airbyte-ready.jpg)
```

> ## Set up the TiDB connector

Reviewer: ```suggestion
## Step 2. Set up the TiDB connector
```

Conveniently, the steps are the same for setting TiDB as the source and the destination.

1. Click **Sources** or **Destinations** in the sidebar and choose TiDB type to create a new TiDB connector.

2. Fill in the following parameters:

    - Host: The host domain of TiDB
    - Port: The port of the database
    - Database: The database that you want to sync the data
    - Username: The username to access the database
    - Password: The password of the username

3. Enable **SSL Connection**, and set TLS protocols to **TLSv1.2** or **TLSv1.3** in **JDBC URL Params**.

    > Note:
    >
    > - TiDB Cloud supports TLS connection. You can choose your TLS protocols in **TLSv1.2** and **TLSv1.3**, for example, `enabledTLSProtocols=TLSv1.2`.
    > - If you want to disable TLS connection to TiDB Cloud via JDBC, you need to set useSSL to `false` in JDBC URL Params specifically and close SSL connection, for example, `useSSL=false`.
    > - TiDB Serverless Tier only supports TLS connections.

4. Click **Set up source** or **destination** to complete creating the connector. The following screenshot shows the configuration of TiDB as the source.

> ![img](/media/tidb-cloud/integration-airbyte-parameters.jpg)

Reviewer: ```suggestion
![TiDB source configuration](/media/tidb-cloud/integration-airbyte-parameters.jpg)
```

For more details about the TiDB connector, see [TiDB Source](https://docs.airbyte.com/integrations/sources/tidb) and [TiDB Destination](https://docs.airbyte.com/integrations/destinations/tidb).

> ## Set up the connection

Reviewer: ```suggestion
## Step 3. Set up the connection
```

> After setting up the source and destination, you can build and configure the connection. You can use any combination of sources and destinations, such as TiDB to Snowflake, and CSV file to TiDB.

Reviewer: > You can use any combination of sources and destinations, such as TiDB to Snowflake, and CSV file to TiDB.

这句话感觉应该放在上一节的最后一步，设置完 TiDB 后，设置另一个source（或destination）

Reviewer 2: Updated

The following steps use TiDB as both source and destination. Other connectors may have different parameters.

1. Click **Connections** in the sidebar and then click **New Connection**.
2. Select the previously established source and destination.

> 3. Go to the **Set up** connection panel and create a name for the connection, such as "${source_name} - ${destination-name}".

Reviewer: ```suggestion
3. Go to the **Set up** connection panel and create a name for the connection, such as `${source_name} - ${destination-name}`.
```

4. Set **Replication frequency** to **Every 24 hours** which means the connection replicates data once a day.
5. Set **Destination Namespace** to **Custom format** and set **Namespace Custom Format** to **test** to store all data in the `test` database.
6. Choose the **Sync schema** to **Full refresh | Overwrite**.

>     > **Tips**

Reviewer: ```suggestion
    > **Tip:**
```

    >

>     > The TiDB connector supports both Incremental and Full Refresh syncs. In Incremental mode, Airbyte only reads records added to the source since the last sync job. The first sync using Incremental mode is equivalent to Full Refresh mode. In Full Refresh mode, Airbyte reads all records in the source and replicates to the destination in every sync task. You can set the sync mode for every table named **Namespace** in Airbyte individually.

Reviewer: ```suggestion
    > The TiDB connector supports both Incremental and Full Refresh syncs.
    >
    > - In Incremental mode, Airbyte only reads records added to the source since the last sync job. The first sync using Incremental mode is equivalent to Full Refresh mode.
    > - In Full Refresh mode, Airbyte reads all records in the source and replicates to the destination in every sync task. You can set the sync mode for every table named **Namespace** in Airbyte individually.
```

>     ![img](/media/tidb-cloud/integration-airbyte-connection.jpg)

Reviewer: ```suggestion
    ![Set up connection](/media/tidb-cloud/integration-airbyte-connection.jpg)
```

    - Port: The port of the database
    - Database: The database that you want to sync the data
    - Username: The username to access the database
    - Password: The password of the username

3. Enable **SSL Connection**, and set TLS protocols to **TLSv1.2** or **TLSv1.3** in **JDBC URL Params**.

    > Note:
    >
    > - TiDB Cloud supports TLS connection. You can choose your TLS protocols in **TLSv1.2** and **TLSv1.3**, for example, `enabledTLSProtocols=TLSv1.2`.
    > - If you want to disable TLS connection to TiDB Cloud via JDBC, you need to set useSSL to `false` in JDBC URL Params specifically and close SSL connection, for example, `useSSL=false`.
    > - TiDB Serverless Tier only supports TLS connections.

4. Click **Set up source** or **destination** to complete creating the connector. The following screenshot shows the configuration of TiDB as the source.

![TiDB source configuration](/media/tidb-cloud/integration-airbyte-parameters.jpg)

For more details about the TiDB connector, see [TiDB Source](https://docs.airbyte.com/integrations/sources/tidb) and [TiDB Destination](https://docs.airbyte.com/integrations/destinations/tidb).

## Set up the connection

> After setting up the source and destination, you can build and configure the connection. You can use any combination of sources and destinations, such as TiDB to Snowflake, and CSV file to TiDB.

Reviewer 2: ```suggestion
After setting up the source and destination, you can build and configure the connection.
```

---
title: Integrate TiDB Cloud with Airbyte
summary: Learn how to use Airbyte TiDB connector.
---

# Integrate TiDB Cloud with Airbyte

[Airbyte](https://airbyte.com/) is an open-source data integration engine to build Extract, Load, Transform (ELT) pipelines and consolidate your data in your data warehouses, data lakes, and databases. This document describes how to connect Airbyte to TiDB Cloud as a source or a destination.

## Deploy Airbyte

You can deploy Airbyte locally with only a few steps.

1. Install [Docker](https://www.docker.com/products/docker-desktop) on your workspace.

2. Clone the Airbyte source code.

    ```shell
    git clone https://github.com/airbytehq/airbyte.git && \
    cd airbyte
    ```

3. Run the Docker images by docker-compose.

    ```shell
    docker-compose up
    ```

> Once you see an Airbyte banner, you can go to <http://localhost:8000> to visit the UI.

Reviewer: Please give the username (`airbyte`) and password (`password`)

> Once you see an Airbyte banner, you can go to <http://localhost:8000> to visit the UI.

Reviewer 2: ```suggestion
Once you see an Airbyte banner, you can go to <http://localhost:8000> with the username ("airbyte") and password ("password") to visit the UI.
```

> 

Author: It's buried in the logs. You can find banner in your logs.

![Airbyte banner](/media/tidb-cloud/integration-airbyte-ready.jpg)

## Set up the TiDB connector

Conveniently, the steps are the same for setting TiDB as the source and the destination.

1. Click **Sources** or **Destinations** in the sidebar and choose TiDB type to create a new TiDB connector.

2. Fill in the following parameters:

>     - Host: The host domain of TiDB

Reviewer: 对 TiDB Cloud 来说，host 是这个 endpoint 吗？这里是否需要说明？

<img width="482" alt="image" src="https://user-images.githubusercontent.com/59654584/198528941-835d6139-64c2-4999-bdff-04cdbab2b6d0.png">

>     - Host: The host domain of TiDB

Author: ```suggestion
    - Host: The host domain (for TiDB Cloud, use Endpoint)
```

    - Port: The port of the database
    - Database: The database that you want to sync the data
    - Username: The username to access the database
    - Password: The password of the username

3. Enable **SSL Connection**, and set TLS protocols to **TLSv1.2** or **TLSv1.3** in **JDBC URL Params**.

    > Note:
    >
    > - TiDB Cloud supports TLS connection. You can choose your TLS protocols in **TLSv1.2** and **TLSv1.3**, for example, `enabledTLSProtocols=TLSv1.2`.
    > - If you want to disable TLS connection to TiDB Cloud via JDBC, you need to set useSSL to `false` in JDBC URL Params specifically and close SSL connection, for example, `useSSL=false`.
    > - TiDB Serverless Tier only supports TLS connections.

4. Click **Set up source** or **destination** to complete creating the connector. The following screenshot shows the configuration of TiDB as the source.

![TiDB source configuration](/media/tidb-cloud/integration-airbyte-parameters.jpg)

> You can use any combination of sources and destinations, such as TiDB to Snowflake, and CSV file to TiDB.

Reviewer 2: ```suggestion
You can use any combination of sources and destinations, such as TiDB to Snowflake, and CSV files to TiDB.
```

For more details about the TiDB connector, see [TiDB Source](https://docs.airbyte.com/integrations/sources/tidb) and [TiDB Destination](https://docs.airbyte.com/integrations/destinations/tidb).

## Set up the connection

After setting up the source and destination, you can build and configure the connection.

> The following steps use TiDB as both source and destination. Other connectors may have different parameters.

Reviewer: ```suggestion
The following steps use TiDB as both a source and a destination. Other connectors may have different parameters.
```

1. Click **Connections** in the sidebar and then click **New Connection**.
2. Select the previously established source and destination.
3. Go to the **Set up** connection panel and create a name for the connection, such as `${source_name} - ${destination-name}`.

> 4. Set **Replication frequency** to **Every 24 hours** which means the connection replicates data once a day.

Reviewer: ```suggestion
4. Set **Replication frequency** to **Every 24 hours**, which means the connection replicates data once a day.
```

    ![Set up connection](/media/tidb-cloud/integration-airbyte-connection.jpg)

7. Set **Normalization & Transformation** to **Normalized tabular data** to use the default normalization mode, or you can set the dbt file for your job. For more information about normalization, refer to [Transformations and Normalization](https://docs.airbyte.com/operator-guides/transformation-and-normalization/transformations-with-dbt).
8. Click **Set up connection**.
9. Once the connection is established, click **ENABLED** to activate the synchronization task. You can also click **Sync now** to sync immediately.

![Sync data](/media/tidb-cloud/integration-airbyte-sync.jpg)

## Limitations

- The TiDB connector does not support the Change Data Capture (CDC) feature.
- TiDB destination converts the `timestamp` type to the `varchar` type in default normalization mode. It happens because Airbyte converts the timestamp type to string during transmission, and TiDB does not support `cast ('2020-07-28 14:50:15+1:00' as timestamp)`.

> - For some large ETL missions, you need to increase the parameters of [transaction restrictions](/develop/dev-guide-transaction-restraints.md#large-transaction-restrictions) in TiDB.

Reviewer: ```suggestion
- For some large ELT missions, you need to increase the parameters of [transaction restrictions](/develop/dev-guide-transaction-restraints.md#large-transaction-restrictions) in TiDB.
```
Please confirm whether it's ETL or ELT in this case. As far as i know, they are alternatives.

Reviewer 2: We use ETL https://docs.pingcap.com/tidb/dev/overview#use-cases

Author: Airbyte introduces itself as ELT, so we should use ELT here.

> - For some large ETL missions, you need to increase the parameters of [transaction restrictions](/develop/dev-guide-transaction-restraints.md#large-transaction-restrictions) in TiDB.

Reviewer 2: ```suggestion
- For some large ETL (Extract, Transform, Load) missions, you need to increase the parameters of [transaction restrictions](/develop/dev-guide-transaction-restraints.md#large-transaction-restrictions) in TiDB.
```

> - For some large ETL (Extract, Transform, Load) missions, you need to increase the parameters of [transaction restrictions](/develop/dev-guide-transaction-restraints.md#large-transaction-restrictions) in TiDB.

Reviewer 2: ```suggestion
- For some large ELT missions, you need to increase the parameters of [transaction restrictions](/develop/dev-guide-transaction-restraints.md#large-transaction-restrictions) in TiDB.
```
